include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -416,6 +416,50 @@ __get_accessor_ptr(const _Acc& __acc)
 #endif
 }
 
+class __event
+{
+    sycl::event __m_event;
+    sycl::queue __m_queue;
+    bool __in_order = false;
+
+  public:
+    __event(sycl::event __e): __m_event(__e) {}
+    __event() {}
+
+    __event(sycl::queue __q): __m_queue(__q), __in_order(true) {}
+
+    void wait()
+    {
+        if(__in_order)
+            __m_queue.wait_and_throw();
+        else
+            __m_event.wait_and_throw();
+    }
+
+    bool has_event() const { return !__in_order; }
+    operator sycl::event() const 
```

Comment:
Would it be safer to make this an explicit function call like `get_sycl_event()` rather than a conversion function?

I'm worried we might get ourselves into a bit of trouble with a conversion function since this is not always interchangeable with a `sycl::event` (as evidenced by the assert).

Is this only used in the `__future` implementation within this PR currently?